kernel/src/cpu/tlb.rs

```diff
@@ -50,6 +52,28 @@ pub fn flush_tlb_global_sync() {
     do_tlbsync();
 }
 
+pub fn flush_tlb_global_percpu() {
+    let cr4 = read_cr4();
+    let cr4_nopge = {
+        // bitflags requires a mut object to add/remove flags, and bitflags
+        // does not implement either Clone or Copy.
+        let mut temp = CR4Flags::from_bits_truncate(cr4.bits());
+        temp.toggle(CR4Flags::PGE);
+        temp
+    };
+    write_cr4(cr4_nopge);
+    write_cr4(cr4);
```

Comment:
We should derive `Copy` (and `Clone`) for `CR4Flags` so that we can do:

```suggestion
    let cr4 = read_cr4();
    write_cr4(cr4 ^ CR4Flags::PGE);
    write_cr4(cr4);
```

Reply:
i.e. simply doing:

```diff
diff --git a/kernel/src/cpu/control_regs.rs b/kernel/src/cpu/control_regs.rs
index 741ea5c..73df73e 100644
--- a/kernel/src/cpu/control_regs.rs
+++ b/kernel/src/cpu/control_regs.rs
@@ -110,6 +110,7 @@ pub fn write_cr3(cr3: PhysAddr) {
 }
 
 bitflags! {
+    #[derive(Debug, Clone, Copy)]
     pub struct CR4Flags: u64 {
         const VME       = 1 << 0;  // Virtual-8086 Mode Extensions
         const PVI       = 1 << 1;  // Protected-Mode Virtual Interrupts

```

Reply:
That is way, way better.  I didn't realize it would be that simple.